include/caffe/datum_DB_factory.hpp

```diff
@@ -0,0 +1,98 @@
+#ifndef CAFFE_DATUMDB_FACTORY_H_
+#define CAFFE_DATUMDB_FACTORY_H_
+
+#include <map>
+#include <string>
+
+#include "caffe/common.hpp"
+#include "caffe/proto/caffe.pb.h"
+
+namespace caffe {
+
+class DatumDB;
+
+class DatumDBRegistry {
+ public:
+  typedef DatumDB* (*Creator)(const DatumDBParameter&);
+  typedef std::map<string, Creator> CreatorRegistry;
+  typedef std::map<string, DatumDB* > SourceRegistry;
+
+  static CreatorRegistry& Registry() {
+    static CreatorRegistry* g_registry_ = new CreatorRegistry();
+    return *g_registry_;
+  }
+
+  static SourceRegistry& Sources() {
```

Comment:
Yeah, there's nothing wrong with using `static` members in non-factory classes where appropriate (which is admittedly rare).